services/vmalert/alerting_rules.go

```diff
@@ -53,19 +59,52 @@ func (s *AlertingRules) ValidateRules(ctx context.Context, rules string) error {
 
 // ReadRules reads current rules from FS.
 func (s *AlertingRules) ReadRules() (string, error) {
-	b, err := ioutil.ReadFile(alertingRulesFile)
-	if err != nil && !os.IsNotExist(err) {
-		return "", err
-	}
-	return string(b), nil
+	// TODO: this method supposed to return external alertmanager rules.
+	// TODO: For now we have common rules for both, external and internal alertmanagers.
+	return "", nil
 }
 
 // RemoveRulesFile removes rules file from FS.
 func (s *AlertingRules) RemoveRulesFile() error {
-	return os.Remove(alertingRulesFile)
+	// TODO: same as ReadRules()
+	return nil
 }
 
 // WriteRules writes rules to file.
 func (s *AlertingRules) WriteRules(rules string) error {
-	return ioutil.WriteFile(alertingRulesFile, []byte(rules), 0o644) //nolint:gosec
+	// TODO: same as ReadRules()
+	return nil
+}
+
+// GetRulesHash returns current rules files hash sum.
+func (s *AlertingRules) GetRulesHash() ([]byte, error) {
```

Comment:
Let's drop the whole thing: Prometheus had a problem when it was reloaded too often with a large number of scrape targets, that's why we check if Prometheus config really changed; VMAlert does not have this problem.

Reply:
Droped